workloads/vm.go

```diff
@@ -226,8 +100,10 @@ func (v VM) Convert(manager deployer.DeploymentManager) {
 	workloads = append(workloads, workload)
 
 	for _, w := range workloads {
-		manager.SetWorkload(v.NodeId, w)
+		err = manager.SetWorkload(v.NodeId, w)
```

Comment:
i think this way err is assigned to the last SetWorkload call.
maybe call SetWorkload on every generated workload and check for error.